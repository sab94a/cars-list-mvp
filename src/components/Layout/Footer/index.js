//@flow

import React from 'react';
import type { Node } from 'react';
import styles from './index.module.scss';

type Props = {
    children: Node
}

const Footer = ({ children }: Props) => (
    <footer className={ styles.root }>{ children }</footer>
);

export default Footer;
