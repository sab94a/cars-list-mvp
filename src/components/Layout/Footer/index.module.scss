@import 'components/lib/variables.scss';

.root {
    display: flex;
    border-top: $border;
    height: $panel-height + $border-width;
    align-items: center;
    justify-content: center;
}
