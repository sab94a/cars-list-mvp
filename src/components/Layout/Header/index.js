//@flow

import React from 'react';
import { Link } from 'react-router-dom';
import Navigation from 'components/lib/Navigation';
import Logo from 'components/lib/Logo';
import type { NavItem } from 'components/lib/types';
import styles from './index.module.scss'

type Props = {
    navigation: Array<NavItem>
};

const Header = ({ navigation }: Props) => (
    <header className={ styles.root }>
        <Link className={ styles.logo } to='/'>
            <Logo />
        </Link>
        <Navigation items={ navigation } />
    </header>
);

export default React.memo<Props>(Header);
