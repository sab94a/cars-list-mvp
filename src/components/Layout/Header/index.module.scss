@import 'components/lib/variables.scss';

.root {
    display: flex;
    justify-content: space-between;
    border-bottom: $border;
}

.logo {
    display: flex;
    align-items: center;
    padding: 0 $offset-lg;
    height: $panel-height;
}