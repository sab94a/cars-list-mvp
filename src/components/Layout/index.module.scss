.root {
    min-height: 100%;
    display: flex;
    flex-direction: column;
}

.content {
    flex-grow: 1;
    display: flex;
}
