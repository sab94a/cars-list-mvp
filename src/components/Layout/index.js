//@flow

import React from 'react';
import type { Node } from 'react';
import type { NavItem } from 'components/lib/types';
import Header from './Header';
import Footer from './Footer';
import styles from './index.module.scss';

export type Props = {
    children: Node,
    navigation: Array<NavItem>,
    footerText: string
};

const Layout = ({ children, navigation, footerText }: Props) => (
    <div className={ styles.root }>
        <Header navigation={ navigation } />
        <main className={ styles.content }>{ children }</main>
        <Footer>{ footerText }</Footer>
    </div>
);

export default Layout;
