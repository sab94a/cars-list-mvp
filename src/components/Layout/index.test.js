import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import Header from './Header';
import Footer from './Footer';
import Layout from './';

describe('<Layout>', () => {
    const navigation = [{ href: '1', title: '1'}];
    const footerText = 'footerText';
    const children = 'text';

    let component = null;

    beforeEach(() => {
        component = shallow(
            <Layout navigation={ navigation } footerText={ footerText }>
                { children }
            </Layout>
        );
    });

    it('Should render correctly', () => {
        expect(toJson(component)).toMatchSnapshot();
    });

    it('Should render Header correctly', () => {
        const nav = component.find(Header);

        expect(nav).toHaveLength(1);
        expect(nav.prop('navigation')).toEqual(navigation);
    });

    it('Should render Footer correctly', () => {
        const footer = component.find(Footer);
        
        expect(footer).toHaveLength(1);
        expect(footer.prop('children')).toEqual(footerText);
    });

    it('Should render child content correctly', () => {
        const main = component.find('main');

        expect(main).toHaveLength(1);
        expect(main.text()).toEqual(children);
    });
});
