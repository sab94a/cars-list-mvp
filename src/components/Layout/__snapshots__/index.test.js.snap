// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Layout> Should render correctly 1`] = `
<div
  className="root"
>
  <Memo(Header)
    navigation={
      Array [
        Object {
          "href": "1",
          "title": "1",
        },
      ]
    }
  />
  <main
    className="content"
  >
    text
  </main>
  <Footer>
    footerText
  </Footer>
</div>
`;
